lib/elixir/lib/map.ex

```diff
@@ -1149,6 +1149,46 @@ defmodule Map do
     end
   end
 
+  @doc """
+  Intersects two maps into one.
+
+  If a key exists in both maps, the value in `map2` will be used.
+
+  Inlined by the compiler.
+
+  ## Examples
+
+      iex> Map.intersect(%{a: 1, b: 2}, %{b: "b", c: "c"})
+      %{b: "b"}
+
+  """
+  @doc since: "1.15.0"
+  @spec intersect(map, map) :: map
+  defdelegate intersect(map1, map2), to: :maps
+
+  @doc """
+  Intersects two maps into one, resolving conflicts through the given `fun`.
+
+  The given function will be invoked when there are duplicate keys; its
+  arguments are `key` (the duplicate key), `value1` (the value of `key` in
+  `map1`), and `value2` (the value of `key` in `map2`). The value returned by
+  `fun` is used as the value under `key` in the resulting map.
+
+  Inlined by the compiler.
```

Comment:
This is not the case, right?

Reply:
What exactly?

> `map1`), and `value2` (the value of `key` in `map2`). The value returned by
  `fun` is used as the value under `key` in the resulting map.

I tried to adapt the same wording used in `Map.merge`, akin to https://www.erlang.org/doc/man/maps.html#intersect_with-3.

> Inlined by the compiler.

I cargo-culted this since other functions that just pass to Erlang functions included it (even without `defdelegate`). E.g. https://github.com/elixir-lang/elixir/blob/bc0cf0b0f5dddec65e1305f4d7906064a91f6c29/lib/elixir/lib/map.ex#L275 or https://github.com/elixir-lang/elixir/blob/bc0cf0b0f5dddec65e1305f4d7906064a91f6c29/lib/elixir/lib/map.ex#L388. But I'm not entirely sure, happy to remove

Reply:
AFAIK you'd have to add the inline rules at https://github.com/elixir-lang/elixir/blob/main/lib/elixir/src/elixir_rewrite.erl, which I believe is expected in this case.

Reply:
I would prefer documentation use language based on the Elixir `MapSet` documentation rather than the Erlang `:maps` docs. For example `MapSet.intersection/2` docs include
>Returns a set containing only members that map_set1 and map_set2 have in common.

This is much nicer than 
>Intersects two sets into one.